packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -16,10 +16,12 @@ import { getSurfacePath } from "./ALSurfaceUtils";
 import { ALFlowletEvent, ALReactElementEvent, ALSharedInitOptions, ALTimedEvent } from "./ALType";
 
 export type ALUIEvent = Readonly<{
-  event: string,
-  element: Element,
+  domEvent: GlobalEventHandlersEventMap[ALUIEvent['event']],
```

Comment:
This might slowdown the compiler. Why not just `keyof GlobalEventHandlersEventMap`?

Reply:
I was hoping there was a way to narrow down to the specific event type based on the actual event in the object but I think it's a bit too clever.  e.g. If event = `keyup` the domEvent would be `KeyboardEvent` typed.

Reply:
there is that possibility if you break this type into a bunch of unions for example:
```
{
event: 'click' | 'mouseover' | 'mousedown'; 
domEvent: GlobalEventHandlersEventMap[ALUIEvent['event']]; // or just MouseEvent
} | {
event: 'keydown' | 'keyup' | 'keypress'; 
domEvent: GlobalEventHandlersEventMap[ALUIEvent['event']]; // or just KeyboardEvent
} | {
event: string;
domEvent: GlobalEventHandlersEventMap[ALUIEvent['event']]; // or just Event
}
```

Reply:
Oh good idea.  Even though it's a bit tedious and ugly I think it will be more clean on the subscriber's end to not have to do instance of assertions every place they want to inspect the event attributes.

Reply:
This actually gets me very close to what I want but when I try to use `eventFilter` later on in the listener it wants to account for every type of event
```ts
type UIEventConfigMapping<T> = {[K in keyof T]: {
  readonly eventName: K,
  // A callable filter for this event, returning true if the event should be emitted, or false if it should be discarded up front
  eventFilter?: (domEvent: T[K]) => boolean;
  // Whether to limit to elements that are "interactable", e.g. have event handlers registered to the element.  Defaults to true.
  interactableElementsOnly?: boolean;
  // Whether to cache element's react information on capture, defaults to false.
  cacheElementReactInfo?: boolean;
}}[keyof T];

type UIEventConfig = UIEventConfigMapping<GlobalEventHandlersEventMap>;
```
<img width="935" alt="image" src="https://user-images.githubusercontent.com/6064408/228369222-1297742a-fae5-45bf-baa3-52c9a1cfc8d4.png">

Is manually unioning the permutations the only way to solve that?

Reply:
no, union is not the only way, the way you're approaching it is actually good (better). Assuming you do trust the `document.addEventListener` to give you the right type, you may have to cast the step to get around the compile error. That might be a reasonable compromise in this case.

Reply:
Pushed up a new commit that "better" types the ALUIEvent,  so that in subscribers the domEvent can be narrowed by checking event is X (`click`,  `keydown` etc...).   Also included this change in the UIEventConfig, so that in configuration the type is stricter for the filter function defined.

All this at the expense of a couple `as any` casts I seemingly (might be a way around this) have to do to satisfy the compiler when we are looping through event config and processing the union of dom event types.  Let me know if you think this is better or worse,  I know there's tradeoffs but maybe I missed something.   I think this makes it a bit easier for usage in AutoLogging.init and also in the subscribers to al_ui_event.